src/Query.php

```diff
@@ -0,0 +1,468 @@
+<?php
+
+namespace Muffin\Webservice;
+
+use Cake\Datasource\Exception\RecordNotFoundException;
+use Cake\Datasource\QueryInterface;
+use Cake\Datasource\QueryTrait;
+use Cake\Utility\Hash;
+use Muffin\Webservice\Model\Endpoint;
+use Muffin\Webservice\Webservice\WebserviceInterface;
+
+class Query implements QueryInterface
+{
+
+    use QueryTrait;
+
+    const ACTION_CREATE = 1;
+    const ACTION_READ = 2;
+    const ACTION_UPDATE = 3;
+    const ACTION_DELETE = 4;
+
+    /**
+     * Indicates that the operation should append to the list
+     *
+     * @var int
+     */
+    const APPEND = 0;
+
+    /**
+     * Indicates that the operation should prepend to the list
+     *
+     * @var int
+     */
+    const PREPEND = 1;
+
+    /**
+     * Indicates that the operation should overwrite the list
+     *
+     * @var bool
+     */
+    const OVERWRITE = true;
+
+    /**
+     * Indicates whether internal state of this query was changed, this is used to
+     * discard internal cached objects such as the transformed query or the reference
+     * to the executed statement.
+     *
+     * @var bool
+     */
+    protected $_dirty = false;
+
+    private $_action;
+    private $_conditions = [];
+    private $_page;
+    private $_limit;
+    private $_fields = [];
+    private $_offset = [];
+    private $_order = [];
```

Comment:
Why `private`? While I like from an OO aspect, in OSS sometimes it makes things harder...